libr/util/sys.c

```diff
@@ -363,26 +363,35 @@ R_API int r_sys_crash_handler(const char *cmd) {
 
 	free (crash_handler_cmd);
 	crash_handler_cmd = strdup (cmd);
-	sigact.sa_handler = signal_handler;
-	sigemptyset (&sigact.sa_mask);
-	sigact.sa_flags = 0;
-	sigaddset (&sigact.sa_mask, SIGINT);
-	sigaddset (&sigact.sa_mask, SIGSEGV);
-	sigaddset (&sigact.sa_mask, SIGBUS);
-	sigaddset (&sigact.sa_mask, SIGQUIT);
-	sigaddset (&sigact.sa_mask, SIGHUP);
+	sigemptyset (&mask);
+	sigaddset (&mask, SIGINT);
+	sigaddset (&mask, SIGSEGV);
+	sigaddset (&mask, SIGBUS);
+	sigaddset (&mask, SIGQUIT);
+	sigaddset (&mask, SIGHUP);
 
-	sigaction (SIGINT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGSEGV, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGBUS, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGQUIT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGHUP, &sigact, (struct sigaction *)NULL);
+	r_sys_sigaction (SIGINT, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGSEGV, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGBUS, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGQUIT, signal_handler, &mask, 0);
+	r_sys_sigaction (SIGHUP, signal_handler, &mask, 0);
 	return true;
 #else
 	return false;
 #endif
 }
 
+#if __UNIX__
```

Comment:
why cover this function under unix with this and then cover all the other calls ot this function too?

make this if cover only the contents of the function and then call it always, also, make that use signal() when sigaction is not available

Reply:
Actually, sigaction is already used directly in master libr/util/sys.c, proving we don't need more preprocessor branching.

Reply:
Where? Could you past a link to the line in the file?

Reply:
https://github.com/radare/radare2/blob/57024df5f318ad3e1f4735cd6493f75c50ca1620/libr/util/sys.c#L375

Reply:
That line is still subject to `#if __UNIX__` at https://github.com/radare/radare2/blob/57024df5f318ad3e1f4735cd6493f75c50ca1620/libr/util/sys.c#L353